.travis.yml

```diff
@@ -11,24 +11,6 @@ env:
 
 before_install:
 
-  # TODO: undo
-  # remove submodules, we don't want those to be actually tested for compliance
-  - git submodule status | rm -Rf `cut -d ' ' -f 3`
-
-  # Formatting checks:
```

Comment:
Hi @tobozo. Just out of an interest in feedback on these, was there a problem that required the formatting checks to be removed, or did they just not end being useful to this project?

Reply:
Hi :-)

As I'm not the maintainer of this project the code quality level is @justcallmekoko's choice.

I just removed those tests because the travis build was [failing](https://travis-ci.com/github/justcallmekoko/ESP32Marauder/builds/176185283), and also because removing the bash lines was faster than finding how to prune xbm/pdf plus fixing the actual formatting errors in the code :-)

some thoughts:
  - An allow-list of file extensions instead of a prune file filter would probably be easier to modify for bash noobs like me
  - Being able to choose between "stop build" and "report only" when formatting checks fail could prevent the clumsy removal of those tests

Reply:
>finding how to prune xbm/pdf

Looks like the entire `pictures/` and `schematics/` folders could just be pruned.

>plus fixing the actual formatting errors in the code

I provide scripts to fix all the formatting issues here:
https://github.com/per1234/formatting-checks/tree/fe641323f3f008333444adc3947bcb1527d0a29a/correction-scripts

>An allow-list of file extensions

There's a demonstration of how to do this here:
https://github.com/per1234/formatting-checks/blob/fe641323f3f008333444adc3947bcb1527d0a29a/.travis.yml#L40-L41

>Being able to choose between "stop build" and "report only" when formatting checks fail

You should put the formatting checks in a separate CI job from the compilation tests, like this:
https://github.com/per1234/arduino-ci-script/blob/1.3.2/.travis.yml#L95-L112